### PR TITLE
WIP: introduce retries to Delete.cs

### LINKED ISSUE #199: adding Retries, RetryDelayMilliseconds parameters to Delete task
Copy task already has such parameters.
it is good for situations when one asynchronously stops windows service (for example) and one need to make sure that files are not locked anymore

I'm not saying that situation itself is good. However, I like the approach of "Copy" task, it allows me to wait for some time and copy files again.

I would like to add such parameters to "Delete" task as well.

if nobody minds, I will make pull request

### PR DESCRIPTION
**Pull request by melytc**
Fixes #199.

Missing: test this.

### CODE REVIEW

File: src/Tasks/Delete.cs [hunk 1]
```diff
@@ -62,45 +67,57 @@ public override bool Execute()
         {
             var deletedFilesList = new List<ITaskItem>();
             var deletedFilesSet = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            int retries = 1;
```
Comment (Forgind) [inline src/Tasks/Delete.cs:70]: This should be in the foreach loop.

**Review by Forgind — APPROVED** [review]
Maybe a test?

**Comment by Forgind** [comment]
Microsoft.Build.UnitTests.Evaluation.Expander_Tests.TestItemSpecModiferEscaping seems to be hanging.

**Comment by Forgind** [comment]
Going to have a chance to get back to this?

**Comment by rainersigwald** [comment]
I'm going to close this for now. We'd love to reopen if you get back to it.

**Pull request closed without merging**